app/celery/tasks.py

```diff
@@ -644,18 +644,11 @@ def process_incomplete_job(job_id):
 
     csv = get_recipient_csv(job, template)
     rows = csv.get_rows()  # This returns an iterator
-    first_row = []
-    for row in rows:
-        if row.index > resume_from_row:
-            first_row.append(row)
-            for result in chunked(rows, Config.BATCH_INSERTION_CHUNK_SIZE):
-                result = first_row + result
-                process_rows(result, template, job, job.service)
-                put_batch_saving_bulk_created(
-                    metrics_logger, 1, notification_type=db_template.template_type, priority=db_template.process_type
-                )
-                first_row = []
-            break
+    for result in chunked(islice(rows, resume_from_row, None), Config.BATCH_INSERTION_CHUNK_SIZE):
```

Comment:
`islice(rows, resume_from_row, None)` is just like `rows[resume_from_row:]` but for an iterator